python/python-adaguc-server/ogcapi/routeOGCApi.py

```diff
@@ -0,0 +1,945 @@
+    """
+    res = {
+        "crs": [
+            "http://www.opengis.net/def/crs/OGC/1.3/CRS84",
+            "http://www.opengis.net/def/crs/EPSG/0/4326",
+        ],
+        "collections": [],
+        "links": []
+    }
+    res["links"].append(make_link(".getcollections",
+                                  "self", "application/json", "Metadata about the feature collections"))
+    res["links"].append(make_link(".getcollections",
+                                  "alternate", "text/html", "Metadata about the feature collections (HTML)", {"f": "html"}))
+    for c in generate_collections():
+        res["collections"].append(getcollection_by_name(c["name"]))
+
+    if "f" in request.args and request.args["f"] == "html":
+        response = render_template("collections.html", collections=res)
+        return response
+
+    return res
+
+
+@routeOGCApi.route("/collections/<coll>", methods=["GET"])
+@cross_origin()
+def getcollection(coll):
+    """Collections endpoint.
+    ---
+    get:
+        description: Get collection info
+        parameters:
+            - in: path
+              schema: CollectionParameter
+        responses:
+            200:
+              description: CollectionInfoSchema
+    """
+    collection = getcollection_by_name(coll)
+    if "f" in request.args and request.args["f"] == "html":
+        response = render_template("collection.html", collection=collection)
+        return response
+
+    return collection
+
+
+def request_by_id(url, name, headers=None, requested_id=None):
+    """
+    request_by_id
+    """
+    url = make_wms1_3(
+        url)+"&request=getPointValue&INFO_FORMAT=application/json"
+
+    if requested_id is not None:
+        # Get feature data for this id
+        terms = requested_id.split(";")
+        observedPropertyName = terms[1]
+        url = f"{url}&LAYERS={observedPropertyName}"
+        lon, lat = terms[2].split(",")
+        for term in terms[3:-1]:
+            dim_name, dim_value = term.split("=")
+            if dim_name.lower() == "reference_time":
+                url = f"{url}&DIM_REFERENCE_TIME={dim_value}"
+            elif dim_name.lower() == "elevation":
+                url = f"{url}&ELEVATION={dim_value}"
+            else:
+                url = f"{url}&DIM_{dim_name}={dim_value}"
+
+        url = f"{url}&X={lon}&Y={lat}&CRS=EPSG:4326"
+        timestring = "/".join(terms[-1].split("$"))
+        url = f"{url}&TIME={timestring}"
+        response = requests.get(url, headers=headers, timeout=TIMEOUT)
+        if response.status_code == 200:
+            try:
+                data = json.loads(response.content.decode(
+                    'utf-8'), object_pairs_hook=OrderedDict)
+            except ValueError:
+                root = fromstring(response.content.decode('utf-8'))
+
+                return 400, root[0].text.strip(), None
+            dat = data[0]
+            item_feature = feature_from_dat(dat, observedPropertyName, name)
```

Comment:
It works, but here the order is `dat, observedPropertyName, name`, while in the function it is  `dat, name, observedPropertyName`